scipy/optimize/zeros.py

```diff
@@ -154,6 +158,11 @@ def newton(func, x0, fprime=None, args=(), tol=1.48e-8, maxiter=50,
         Ignored if `x0` is not scalar.
         *Note: this has little to do with displaying, however,
         the `disp` keyword cannot be renamed for backwards compatibility.*
+    avoid_out_of_domain : bool, optional
```

Comment:
Hmm, is there a reason this behaviour needs to be switchable? Especially because you're changing the default behaviour. What would be a reason to go back to the old behaviour?

Reply:
In my view there would be no reason to go back to the old behavior, I would be happy to remove the switch. Another option would be to make it opt-in first, then opt-out, then remove the switch, but I'm not sure if it's worth the complexity.

Reply:
> In my view there would be no reason to go back to the old behavior, I would be happy to remove the switch. 

That sounds good, would avoid making the API messy.

> Another option would be to make it opt-in first, then opt-out, then remove the switch

That's never a good idea. We tried many flavors of this in the past, but it usually just makes users change their code twice and drags things out for too long.